lib/common/digest.c

```diff
@@ -59,6 +59,10 @@ dump_xml_for_digest(xmlNodePtr xml)
 static char *
 calculate_xml_digest_v1(xmlNode *input)
 {
+    /* Removing v1 patchsets affects even full-restart upgrades from old
+     * versions due to their continued use in digests of operations and on-disk
+     * CIBs. Dropping them is a compatibility nightmare.
+     */
```

Comment:
Operation digests don't matter in a full restart because they're in the status section. On-disk CIBs however are a problem.

Reply:
I wondered about that. I assumed they did and that was why we unconditionally used v1 up to this point.

Reply:
Would switching the operation digest format break rolling upgrades though? At least some state is preserved, and I wonder if the DC would end up computing a different digest format than some newer node and needing to compare them at some point...

Reply:
s/patchsets/digests/ in the comment (two different things)

Operation digests need v1 for the sorting. Otherwise different attribute orders would give different digests. v1/v2 is really sorted/unsorted digest rather than sequential versions of a digest scheme.

Reply:
> s/patchsets/digests/ in the comment (two different things)

Definitely, I had just been in patchset mode for a while.

> Operation digests need v1 for the sorting. Otherwise different attribute orders would give different digests. v1/v2 is really sorted/unsorted digest rather than sequential versions of a digest scheme.

That's not a problem. The sorting happens outside of the v1 function now. We call `pcmk__xe_sort_attrs()`.

If sorting is the only thing we need v1 for in op digests, and it won't cause any "digest has changed" type issues during rolling upgrades, then we can move them to v2.